primitives/runtime/src/bounded/bounded_vec.rs

```diff
@@ -104,10 +104,45 @@ where
 /// A bounded slice.
 ///
 /// Similar to a `BoundedVec`, but not owned and cannot be decoded.
-#[derive(Encode, scale_info::TypeInfo)]
-#[scale_info(skip_type_params(S))]
+#[derive(Encode)]
 pub struct BoundedSlice<'a, T, S>(pub(super) &'a [T], PhantomData<S>);
 
+// This can be replaced with
+// #[derive(scale_info::TypeInfo)]
+// #[scale_info(skip_type_params(S))]
+// again once this issue is fixed in the rust compiler: https://github.com/rust-lang/rust/issues/96956
```

Comment:
Please create an issue for this.

Reply:
So, that we don't forget it.

Reply:
https://github.com/paritytech/substrate/issues/11915, I will also put it in the comment here